src/components/TodoItem.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames';
+import React, { Dispatch, SetStateAction } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  handleToggleTodo: (id: number) => void;
+  onDeleteTodo: (id: number) => void;
+  loadingIds: number[];
+  setLoadingIds: Dispatch<SetStateAction<number[]>>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleToggleTodo,
+  loadingIds,
+  setLoadingIds,
+  onDeleteTodo,
+}) => {
+  const { completed, id, title } = todo;
+  const handleDelete = (todoId: number) => {
+    setLoadingIds(prevIds => [...prevIds, todoId]);
+
+    return onDeleteTodo(todoId);
+  };
```

Comment:
```suggestion
  const handleDelete = (todoId: number) => {
    return onDeleteTodo(todoId);
  };
```
you can move it directly to the onDeleteTodo, and remove the setLoadingIds props at all